packages/itwin/tree-widget/src/tree-widget-react/components/trees/categories-tree/CategoriesTreeButtons.tsx

```diff
@@ -52,17 +54,21 @@ export interface CategoriesTreeHeaderButtonProps extends TreeToolbarButtonProps
  *
  * @public
  */
-export function useCategoriesTreeButtonProps({ viewport }: { viewport: Viewport }): {
+export function useCategoriesTreeButtonProps({ viewport }: { viewport: Viewport | TreeWidgetViewport }): {
   buttonProps: Pick<CategoriesTreeHeaderButtonProps, "categories" | "viewport" | "models">;
   onCategoriesFiltered: (props: { categories: CategoryInfo[] | undefined; models?: Id64Array }) => void;
 } {
   const [filteredCategories, setFilteredCategories] = useState<CategoryInfo[] | undefined>();
   const [filteredModels, setFilteredModels] = useState<Id64Array | undefined>();
-  const categories = useCategories(viewport);
-  const models = useAvailableModels(viewport);
+
+  const treeWidgetViewport = useMemo(() => {
+    return createTreeWidgetViewport(viewport);
```

Comment:
Can we handle this somewhere in components higher up instead of doing it in all these buttons? I think `createTreeWidgetViewport` should be called one time above `Tree` and buttons and all buttons and tree renderers should take `TreeWidgetViewport`.

Reply:
But these changes are in the next branch so we can make breaking changes. 🤔

Reply:
I don't think these buttons should be used separately without tree component.

Reply:
ah, completely forgot this is on `tree-widget/next`... in that case, I'd like us to consider going even further and taking `TreeWidgetViewport` everywhere, including the top level components, and exposing the `createTreeWidgetViewport` through the public API.

Reply:
now that we take `TreeWidgetViewport` everywhere, do we need `createTreeWidgetViewport` to take `Viewport | TreeWidgetViewport`? maybe we could just take `Viewport`?